psydac/linalg/fft.py

```diff
@@ -0,0 +1,226 @@
+from psydac.linalg.basic import LinearOperator, LinearSolver
+from psydac.linalg.stencil import StencilVectorSpace
+from psydac.linalg.kron import KroneckerLinearSolver
+
+import numpy as np
+import scipy.fft as scifft
+import os
+
+class DistributedFFTBase(LinearOperator):
+    """
+    A base class for the distributed FFT, DCT and DST.
+    Internally calls a KroneckerLinearSolver on a solver which just applies the FFT or some other function.
+
+    Parameters
+    ----------
+    space : StencilVectorSpace
+        The vector space needed for the KroneckerLinearSolver internally.
+    
+    function : Callable
+        A callable function with one parameter x which applies some function in-place on x.
+    """
+
+    # Possible additions for the future:
+    # * split off the LinearSolver class when used with the space ndarray (as used in the KroneckerLinearSolver),
+    #   and make it state if it works in-place (or if it needs temporary memory), and what its optimal
+    #   size is (FFT might work faster with padding)
+    # * include FFTW support (e.g. pyfftw)
+
+    class OneDimSolver(LinearSolver):
+        """
+        A one-dimensional solver which just applies a given function.
+
+        Parameters
+        ----------
+        function : Callable
+            The given function.
+        """
+        def __init__(self, function):
+            self._function = function
+
+        @property
+        def space(self):
+            return np.ndarray
+        
+        def solve(self, rhs, out=None, transposed=False):
+            if out is None:
+                out = np.empty_like(rhs)
+            
+            if out is not rhs:
+                out[:] = rhs
+            
+            self._function(out)
+
+            return out
+
+    def __init__(self, space, function):
+        assert isinstance(space, StencilVectorSpace)
+        onedimsolver = DistributedFFTBase.OneDimSolver(function)
+        self._isolver = KroneckerLinearSolver(space, [onedimsolver] * space.ndim)
```

Comment:
Here you are using the same 1D solver (`function`) along every direction. This will not work in the future if we use an FFT library where the plan can be precomputed, in which case this will be different for each direction. Given that this is the base class in this new module, we may want to be more careful.

Reply:
I agree. Thanks for this modification, that's sufficient for now!